index.js

```diff
@@ -0,0 +1,123 @@
+const http = require('node:http');
+const fs = require('node:fs');
+const url = require('node:url');
+
+const PORT = 3000;
+
+// const todos = fs.readFile('./todos.json', 'utf8', (error, result) => {
+//   if (error) throw error;
+//   console.log(JSON.parse(result));
+//   return JSON.parse(result);
+// });
+
+let todos = [
+  {
+    id: 1,
+    title: 'Title 1',
+    completed: false,
+  },
+  {
+    id: 2,
+    title: 'Title 2',
+    completed: false,
+  },
+  {
+    id: 3,
+    title: 'Title 3',
+    completed: false,
+  },
+  {
+    id: 4,
+    title: 'Title 4',
+    completed: false,
+  },
+];
+
+const server = http.createServer((req, res) => {
+  const serverURL = url.parse(req.url, true);
+  const searchId = parseInt(serverURL.query.id);
+  const searchedItem = todos[searchId - 1];
+  const searchIndex = todos.findIndex((item) => item.id === searchId);
+  const firstItemId = todos[0].id;
+  const lastItemId = todos.length;
+  const currentId = lastItemId + 1;
+  const responseHead = { 'Content-Type': 'text/plain' };
```

Comment:
Since you would like to have json result to be sent to client, then change to content type to reflect json instead

Reply:
Fixed